pkg/node/controller.go

```diff
@@ -64,57 +76,97 @@ func (h *handler) updateHosts(node *core.Node, removed bool) (*core.Node, error)
 		}
 	}
 	if h.modCoreDNS {
-		if err := h.updateCoreDNSConfigMap(nodeName, nodeAddress, removed); err != nil {
+		if err := h.updateCoreDNSConfigMap(nodeName, hostName, nodeIPv4, nodeIPv6, removed); err != nil {
 			return nil, err
 		}
 	}
 	return nil, nil
 }
 
-func (h *handler) updateCoreDNSConfigMap(nodeName, nodeAddress string, removed bool) error {
-	if nodeAddress == "" && !removed {
-		logrus.Errorf("No InternalIP found for node " + nodeName)
+func (h *handler) updateCoreDNSConfigMap(nodeName, hostName, nodeIPv4, nodeIPv6 string, removed bool) error {
+	if removed {
+		nodeIPv4 = ""
+		nodeIPv6 = ""
+	} else if nodeIPv4 == "" && nodeIPv6 == "" {
+		logrus.Errorf("No InternalIP addresses found for node " + nodeName)
 		return nil
```

Comment:
Not related to changes here, but shouldn't we return the error instead of just logging it?

Reply:
no, returning an error from a handler causes wrangler to reenque the handler and run it again, with the expectation that there is an external problem that needs to be retried. In this case there is nothing to retry - this state cannot be handled, we just need to wait for another update where the node does have IPs set.

Reply:
ok, thanks for the explanation Brad